amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/Lease.java

```diff
@@ -274,6 +281,18 @@ public void childShardIds(@NonNull final Collection<String> childShardIds) {
         this.childShardIds.addAll(childShardIds);
     }
 
+    /**
+     * Set the hash range key for this shard.
+     * @param hashKeyRangeForLease
+     */
+    public void hashKeyRange(final HashKeyRangeForLease hashKeyRangeForLease) {
+        if(this.hashKeyRangeForLease == null) {
```

Comment:
Seems in the constructor, we are doing null check. But in the setter we are not. Shall we keep this consistent?

Reply:
The one in constructor is for optimization and not for safety. Keeping it consistent.